pkg/ruler/remotequerier.go

```diff
@@ -96,39 +94,20 @@ func (c *QueryFrontendConfig) Validate() error {
 }
 
 // DialQueryFrontend creates and initializes a new httpgrpc.HTTPClient taking a QueryFrontendConfig configuration.
-func DialQueryFrontend(cfg QueryFrontendConfig, reg prometheus.Registerer, logger log.Logger) (httpgrpc.HTTPClient, error) {
-	invalidClusterValidation := util.NewRequestInvalidClusterValidationLabelsTotalCounter(reg, "ruler-query-frontend", util.GRPCProtocol)
-	opts, err := cfg.GRPCClientConfig.DialOption(
-		[]grpc.UnaryClientInterceptor{
-			middleware.ClientUserHeaderInterceptor,
-		},
-		nil,
-		util.NewInvalidClusterValidationReporter(cfg.GRPCClientConfig.ClusterValidation.Label, invalidClusterValidation, logger),
-	)
-	if err != nil {
-		return nil, err
-	}
-	opts = append(opts, grpc.WithDefaultServiceConfig(serviceConfig))
-	opts = append(opts, grpc.WithStatsHandler(otelgrpc.NewClientHandler()))
-
-	// nolint:staticcheck // grpc.Dial() has been deprecated; we'll address it before upgrading to gRPC 2.
-	conn, err := grpc.Dial(cfg.Address, opts...)
-	if err != nil {
-		return nil, err
-	}
-	return httpgrpc.NewHTTPClient(conn), nil
+func DialQueryFrontend(cfg QueryFrontendConfig, prometheusHTTPPrefix string, reg prometheus.Registerer, logger log.Logger) (http.RoundTripper, *url.URL, error) {
+	return dialQueryFrontendGRPC(cfg, prometheusHTTPPrefix, reg, logger)
```

Comment:
nit: do we need a whole function in a new file for this? can't we restore the old version?